lib/util/historical-buffer-manager.ts

```diff
@@ -0,0 +1,95 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import assert = require('assert');
+import interfaces = require('../interface');
+
+// Internal class that holds the buffer array, a counter for overflow items and maxBufferLength
+class Buffer<T> {
+    data: T[];
+    overflow: number;
```

Comment:
I think I like this style too, i.e. use initializers with the variable declarations. It's nice to have the defaults with the variables and it's nice when you have multiple constructors.